platform/default/mbgl/storage/offline_database.cpp

```diff
@@ -49,7 +49,8 @@ void OfflineDatabase::ensureSchema() {
             case 1: break; // cache-only database; ok to delete
             case 2: migrateToVersion3(); // fall through
             case 3: migrateToVersion4(); // fall through
```

Comment:
That’s a great point — I’d lean towards removing `migrateToVersion4` and eliminating any possibility of journal weirdness for the vast majority of users.